src/main/resources/modules/menstrual_cycle.json

```diff
@@ -0,0 +1,350 @@
+      "type": "Terminal"
+    },
+    "Female": {
+      "type": "Simple",
+      "remarks": [
+        "About one-in-five Americans use a smart watch or fitness tracker.  \n https://www.pewresearch.org/short-reads/2020/01/09/about-one-in-five-americans-use-a-smart-watch-or-fitness-tracker/"
+      ],
+      "distributed_transition": [
+        {
+          "transition": "Age_0_9",
+          "distribution": 0.1884
+        },
+        {
+          "transition": "Age_10_19",
+          "distribution": 0.2074
+        },
+        {
+          "transition": "Age_20_29",
+          "distribution": 0.2194
+        },
+        {
+          "transition": "Age_30_39",
+          "distribution": 0.2244
+        },
+        {
+          "transition": "Age_40_49",
+          "distribution": 0.2039
+        },
+        {
+          "transition": "Age_50_59",
+          "distribution": 0.2097
+        },
+        {
+          "transition": "Age_60_69",
+          "distribution": 0.2058
+        },
+        {
+          "transition": "Age_70_79",
+          "distribution": 0.1407
+        },
+        {
+          "transition": "Age_80_99",
+          "distribution": 0.0803
+        }
+      ]
+    },
+    "Age_0_9": {
+      "type": "Delay",
+      "direct_transition": "Terminal",
+      "range": {
+        "low": 0,
+        "high": 9,
+        "unit": "years"
+      }
+    },
+    "Age_10_19": {
+      "type": "Delay",
+      "direct_transition": "Endometriosis_Risk",
+      "range": {
+        "low": 10,
+        "high": 19,
+        "unit": "years"
+      }
+    },
+    "Age_20_29": {
+      "type": "Delay",
+      "direct_transition": "Endometriosis_Risk",
+      "range": {
+        "low": 20,
+        "high": 29,
+        "unit": "years"
+      }
+    },
+    "Age_30_39": {
+      "type": "Delay",
+      "direct_transition": "Endometriosis_Risk",
+      "range": {
+        "low": 30,
+        "high": 39,
+        "unit": "years"
+      }
+    },
+    "Age_40_49": {
+      "type": "Delay",
+      "direct_transition": "Endometriosis_Risk",
+      "range": {
+        "low": 40,
+        "high": 49,
+        "unit": "years"
+      }
+    },
+    "Age_50_59": {
+      "type": "Delay",
+      "direct_transition": "Endometriosis_Risk",
+      "range": {
+        "low": 50,
+        "high": 59,
+        "unit": "years"
+      }
+    },
+    "Age_60_69": {
+      "type": "Delay",
+      "direct_transition": "Endometriosis_Risk",
+      "range": {
+        "low": 60,
+        "high": 69,
+        "unit": "years"
+      }
+    },
+    "Age_70_79": {
+      "type": "Delay",
+      "direct_transition": "Terminal",
+      "range": {
+        "low": 70,
+        "high": 79,
+        "unit": "years"
+      }
+    },
+    "Age_80_99": {
+      "type": "Delay",
+      "direct_transition": "Terminal",
+      "range": {
+        "low": 80,
+        "high": 99,
+        "unit": "years"
+      }
+    },
+
+    "Endometriosis_Risk": {
+      "type": "ConditionOnset",
```

Comment:
These "Risk" states either need to have `codes` defined, or they need to be a different state type. As it stands they cause exceptions when running, see the unit tests: https://github.com/synthetichealth/synthea/actions/runs/6397186772/job/17364591481

If you don't intend for any record artifact to be added for this, you probably just want these to be Simple states

Reply:
Completely overhauled the entire module.  Refactored from `Endometriosis_Risk` to simply `Endometriosis`.   Has a matching ConditionEnd.   Unit tests passing.   No crashing recently.